File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        l = r = 0
        while r < len(nums):
            if nums[r] == 0:
                k -= 1
            if k < 0:
                k += nums[l] == 0
                l += 1
            r += 1
        return r - l

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var longestOnes = function(nums, k) {
  let [l, r] = [0, 0];
  
  while (r < nums.length) {
    if (!nums[r++]) k--;
    if (k < 0 && !nums[l++]) k++;
  }
  
  return r - l;
};